src/System.Net.Http/tests/FunctionalTests/HttpCookieProtocolTests.cs

```diff
@@ -316,6 +316,7 @@ private static CookieContainer CreateSingleCookieContainer(Uri uri, string cooki
 
                 using (HttpClient client = new HttpClient(handler))
                 {
+                    client.DefaultRequestHeaders.ConnectionClose = true; // to avoid issues with connection pooling
```

Comment:
If you suspect a WinHTTP bug with connection re-use, it would be good if you could reword these comments to be clearer regarding why the tests are being changed.

i.e. "// WinHTTP has bugs regarding connection pooling, so don't try to reuse the connection."

Reply:
I don't actually know that they're "bugs" per se.  There's an inherent race condition in connection pooling, in that the server could close the connection after the client has already started reusing it.

Reply:
> I wonder if it also would be useful to add a "IsWinHttpHandler" around these test changes.

We could.  At the same time, the server in the test is explicitly closing the connection.  Since there is an inherent race condition here and since this test isn't about testing connection pooling, it seems fine (desirable, even) to make the test as focused and as deterministic as possible.